test/integration/integration_admin_test.cc

```diff
@@ -510,39 +506,50 @@ TEST_P(IntegrationAdminTest, AdminDrainInboundOnlyIdempotent) {
       lookupPort("admin"), "POST", "/drain_listeners?inboundonly&graceful", "",
       downstreamProtocol(), version_);
   EXPECT_TRUE(response->complete());
-  EXPECT_EQ("304", response->headers().getStatusValue());
+  EXPECT_EQ("200", response->headers().getStatusValue());
   EXPECT_EQ("text/plain; charset=UTF-8", contentType(response));
-  EXPECT_EQ("already draining\n", response->body());
+  EXPECT_EQ("OK\n", response->body());
 
+  // This should actually cause a new drain since this one isn't inbound_only
   BufferingStreamDecoderPtr response3 = IntegrationUtil::makeSingleRequest(
-      lookupPort("admin"), "POST", "/drain_listeners?graceful", "",
-      downstreamProtocol(), version_);
+      lookupPort("admin"), "POST", "/drain_listeners?graceful", "", downstreamProtocol(), version_);
```

Comment:
consider moving this to another test since this one above just seems to be testing admin inbound drain idempotent. 

e.g. a test where we do inbound, then all drain and validate inbound_0 drained from the 1st one and outbound_0 drained on second one.

Reply:
One of the benefits of doing this here though is that we validate the listener metric behavior (i.e even though we drain in two directions, it's only counted once)